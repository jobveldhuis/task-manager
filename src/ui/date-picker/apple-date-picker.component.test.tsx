import { fireEvent, render } from "@testing-library/react-native";
import DateTimePicker from "@react-native-community/datetimepicker";
import { BlurView } from "expo-blur";
import { AppleDatePicker } from "./apple-date-picker.component";

describe("AppleDatePicker", () => {
  const handleBackgroundPress = jest.fn();
  const handleChange = jest.fn();
  const picker = (
    <AppleDatePicker
      onBackgroundPress={handleBackgroundPress}
      value={new Date(1695566021000)}
      onChange={handleChange}
      minimumDate={new Date(1695566020000)}
    />
  );

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("should match snapshot", () => {
    expect(render(picker).toJSON()).toMatchSnapshot();
  });

  it("should fire the onChange event after clicking on a date", () => {
    const { UNSAFE_getByType: getByType } = render(picker);
    const renderedPicker = getByType(DateTimePicker);
    fireEvent(renderedPicker, "onChange", {
      type: "set",
      nativeEvent: {
        timestamp: new Date(1695566029000).getTime(),
        utcOffset: 0,
      },
    });

    expect(handleChange).toBeCalled();
    expect(handleChange).toHaveBeenCalledWith({
      type: "set",
      nativeEvent: {
        timestamp: new Date(1695566029000).getTime(),
        utcOffset: 0,
      },
    });
  });

  it("should fire the onBackgroundPress after clicking the blurred background", () => {
    const { UNSAFE_getByType: getByType } = render(picker);
    const background = getByType(BlurView);

    fireEvent(background, "touchEnd");
    expect(handleBackgroundPress).toBeCalled();
  });
});
