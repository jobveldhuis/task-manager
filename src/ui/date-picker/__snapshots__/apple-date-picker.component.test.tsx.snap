// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppleDatePicker should match snapshot 1`] = `
<View
  onTouchEnd={[MockFunction]}
  style={
    [
      {
        "backgroundColor": "transparent",
      },
      {
        "alignItems": "center",
        "alignSelf": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 32,
        "position": "absolute",
        "width": "100%",
        "zIndex": 999,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoBlurView
    blurReductionFactor={4}
    intensity={20}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    tint="light"
  />
  <View
    onStartShouldSetResponder={[Function]}
    onTouchEnd={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "paddingBottom": 32,
        "paddingHorizontal": 16,
        "paddingTop": 24,
        "position": "absolute",
        "top": "25%",
        "width": "100%",
      }
    }
  >
    <RNDateTimePicker
      date={1695566021000}
      displayIOS="inline"
      enabled={true}
      minimumDate={1695566020000}
      mode="date"
      onChange={[Function]}
      onPickerDismiss={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      themeVariant="light"
    />
  </View>
</View>
`;
